PR: Update plugin to work with Spyder 4

bgallois opened the pull request:
Updated plugin to work with Spyder 4, fixes #43

goanpeca commented:
@bgallois thanks for this!

Will review this sometime this week. Thanks again :-)

goanpeca commented on spyder_vim/tests/test_vim.py:
```diff
@@ -24,7 +24,12 @@
+    from spyder.plugins.editor.widgets.editor import EditorStack
+except ImportError:
+    # Spyder 3
+    from spyder.widgets.editor import EditorStack
```
@ccordoba12 should we remove spy3.x compatibility ?

goanpeca commented:
I think we will probably remove Spyder 3 compat, besides that looks good!

goanpeca commented on spyder_vim/tests/test_vim.py:
```diff
@@ -24,7 +24,12 @@
 
 # Spyder imports
 # from spyder.utils.fixtures import setup_editor
```
we can remove this line if we are not using it

goanpeca commented on spyder_vim/tests/test_vim.py:
```diff
@@ -34,6 +39,11 @@
 LOCATION = osp.realpath(osp.join(
     os.getcwd(), osp.dirname(__file__)))
 
+class VimTesting(Vim):
```
Please add a space above here

goanpeca commented on spyder_vim/tests/test_vim.py:
```diff
@@ -34,6 +39,11 @@
+    CONF_FILE = False
+
+    def __init(self, parent):
+        Vim.__init__(self, parent)
```
Please add a space below here

goanpeca commented on spyder_vim/vim.py:
```diff
@@ -42,10 +50,11 @@ def get_plugin_icon(self):
         self.vim_cmd.editor_widget.layout().addWidget(self.vim_cmd)
-        fixed_shortcut("Esc", self.vim_cmd.editor_widget.editorsplitter,
-                       self.vim_cmd.commandline.setFocus)
+        sc = QShortcut("Esc", self.vim_cmd.editor_widget.editorsplitter, self.vim_cmd.commandline.setFocus)
```
Is this "Esc" a translatable string @bgallois ?

bgallois commented:
@goanpeca thanks! I fixed the formatting and the key sequence in the shortcut.
The spyder 3 compatibility is removed from the test and fixed for the plugin.
